refinery/ui/source/js/commons/ctrls/permission-editor-ctrl.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Permission Editor Ctrl
+ * @namespace PermissionEditorCtrl
+ * @desc Main controller for the permission modal component.
+ * @memberOf refineryApp
+ */
+
+(function () {
+  'use strict';
+
+  angular
+    .module('refineryApp')
+    .controller('PermissionEditorCtrl', PermissionEditorCtrl);
+
+  PermissionEditorCtrl.$inject = [
+    '$log',
+    '_',
+    'sharingService',
+    'permissionService'
+  ];
+
+  function PermissionEditorCtrl (
+    $log,
+    _,
+    sharingService,
+    permissionService
+  ) {
+    var vm = this;
+    vm.cancel = cancel;
+    vm.permissions = permissionService.permissions;
+    // Used as a shorthand to avoid complicated permission checking in `ngRepeat`
+    vm.permissionLevel = {
+      none: {
+        read: false,
+        read_meta: false,
+        change: false
+      },
+      read_meta: {
+        read: false,
+        read_meta: true,
+        change: false
+      },
+      read: {
+        read: true,
+        read_meta: true,
+        change: false
+      },
+      edit: {
+        read: true,
+        read_meta: true,
+        change: true
+      }
```

Comment:
consider swapping "read_meta" and "read" in these property lists, so we have the same increasing sequence everywhere?